pkg/dynatrace-client/client.go

```diff
@@ -206,6 +210,52 @@ func (c *client) makeRequest(format string, a ...interface{}) (*http.Response, e
 	return c.httpClient.Get(url)
 }
 
+// PostMarkedForTerminationEvent =>
+// send event to dynatrace api that an event has been marked for termination
+func (c *client) PostMarkedForTerminationEvent(nodeID string) (string, error) {
+
+	url := fmt.Sprintf("%s/v1events", c.url)
+
+	body := `
+	{
+		"eventType": "MARKED_FOR_TERMINATION",
+		"start": %v,
+		"attachRules": {
+		  "entityIds": [
+			%s
+		  ],
+		  "tagRule": [
+			{
+			  "meTypes": [
+				"HOST"
+			  ],
+			  "tags": [
+				{
+				  "context": "CONTEXTLESS",
+				  "key": "customTag"
+				}
+			  ]
+			}
+		  ]
+		},
+		"source": "OpsControl",
+		"annotationType": "defect",
+		"annotationDescription": "The node is marked for termination"
+	  }
+	`
+	bbytes := []byte(fmt.Sprintf(body, time.Now().Unix(), nodeID))
```

Comment:
Do we need to send here the node name, or the Dynatrace's host ID used? If it's the latter, then you may need to query the host ID from the IP. We do something similar for getting the agent version on the host.

Reply:
The doc said, nodeID

Reply:
Whatever it's needed, but I'd be really surprised if it's not the Dynatrace Host Entity ID.